pkg/networkservice/common/discover/match_selector.go

```diff
@@ -55,7 +55,7 @@ func matchEndpoint(nsLabels map[string]string, ns *registry.NetworkService, netw
 		for _, destination := range match.GetRoutes() {
 			// Each NSE should be matched against that destination
 			for _, nse := range networkServiceEndpoints {
-				if isSubset(nse.GetLabels(), destination.GetDestinationSelector(), nsLabels) {
+				if isSubset(nse.GetNetworkServiceLabels()[ns.Name].Labels, destination.GetDestinationSelector(), nsLabels) {
```

Comment:
Did we rename labels in the NSE to be Network Service Labels?

Reply:
Yes, at this moment this map support case multiple network services per single endpoint :)

Reply:
Got it :)